instrumentor/controllers/agentenabled/pods_webhook.go

```diff
@@ -256,10 +280,12 @@ func (p *PodsWebhook) injectOdigosToContainer(containerConfig *odigosv1.Containe
 	}
 
 	volumeMounted := false
+	containerDirsToCopy := make(map[string]struct{})
 	if distroMetadata.RuntimeAgent != nil {
-		if *config.MountMethod == common.K8sHostPathMountMethod {
-			// mount directory only if the mount type is host-path
+		if *config.MountMethod == common.K8sHostPathMountMethod || *config.MountMethod == common.K8sInitContainerMountMethod {
+			// mount directory only if the mount type is host-path or init container
 			for _, agentDirectoryName := range distroMetadata.RuntimeAgent.DirectoryNames {
+				containerDirsToCopy[agentDirectoryName] = struct{}{}
```

Comment:
the `containerDirsToCopy` is only for `init-container` but it will also be populated for `host-path`.

How about calling it `mountedAgentDirectories`, so it describe more precisely what it's role in the context of this function